### PR TITLE
Add features for Gamepad API

### PR DESCRIPTION
**Pull request by queengooborg**
This PR adds three new features related to the Gamepad API.  The haptics-related features were split into their own subfeature, as not all browsers support haptics and the interface has changed for haptic feedback.

### CODE REVIEW

File: features/gamepad-haptics.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: gamepad-haptics
```
Comment (captainbrosset) [inline features/gamepad-haptics.yml:1]:
```suggestion
name: Gamepad haptic feedback
```

File: features/gamepad-haptics.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+description: The gamepad haptics interfaces allow developers to control the vibration motors in gamepads.
```
Comment (captainbrosset) [inline features/gamepad-haptics.yml:2]:
```suggestion
description: The `gamepad.vibrationActuator` property represents a gamepad's haptics hardware, which allows you to control the vibration motors in a gamepad.
```

File: features/gamepad-extensions.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+name: gamepad
+description: The Gamepad Extensions API allows developers to seamlessly interface with gamepad-like devices with advanced capabilities, such as controllers for XR headsets.
```
Comment (captainbrosset) [inline features/gamepad-extensions.yml:2]: Based on the above, I would rephrase to something like this:

```suggestion
description: The `gamepad.hand` and `gamepad.pose` properties of the Gamepad API represent the position in space and the hand that WebXR controllers are being held in.
```

This would make this feature more specific to a developer use case (i.e. making a game compatible with VR headsets), and less about mapping to one spec in particular.

The `GamepadHapticActuator` BCD keys you've added below feel like they belong to the `gamepad-haptics.yml` file instead.

**Review by captainbrosset — CHANGES_REQUESTED** [review]

File: features/gamepad-extensions.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+name: gamepad
```
[the remark this answers is not in the document]
Comment (queengooborg) [reply features/gamepad-extensions.yml]: The spec itself is called the "Gamepad Extensions", so that's why I elected to call it as such.  It is true that these updates are tailored towards XR controllers, but looking at some of these interfaces, I could see them used for a non-XR gamepad.  For example, the DualShock 4 and DualSense controllers have a gyroscope in them, which could be reflected through `GamepadPose.orientation`.  They also have a touchpad, which could be reflected by `GamepadTouch`.

[the remark this answers is not in the document]
Comment (captainbrosset) [reply features/gamepad-extensions.yml]: Got it. Though our overarching goal is to design features around what developers would want to use in their apps, not around how specs are structured. So if a spec contains one set of interfaces for doing X, and another set of interfaces for doing Y, we would create 2 separate features: X and Y.
I think this is likely due to how we generated a bunch of the draft features, which we did by using a script that looped over specs.

From what you said, it seems like there could almost be one feature for gamepad XR headset support, one for gamepad orientation, and one for gamepad touch support.
What do you think?

Also pinging @ddbeck on this one, to help make a decision re splitting this feature.

**Review by ddbeck — CHANGES_REQUESTED** [review]
Line comments on some minor issues, with an overall comment in regards to https://github.com/web-platform-dx/web-features/pull/1835#discussion_r1778644382:

I don't love the "extensions" structure (it reflects the spec authors' view of the world, not developer tasks). I think Patrick's suggestion to break this up ("gamepad XR headset support, one for gamepad orientation, and one for gamepad touch support") into distinct slices of useful gamepad data is the right move. I also couldn't find very many references to "gamepad extensions" that weren't talking about the spec more or less directly, further supporting this route.

File: features/gamepad-extensions.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+name: Gamepad extensions
+description: The `gamepad.hand` and `gamepad.pose` properties of the Gamepad API represent the position in space and the hand that WebXR controllers are being held in.
+spec: https://w3c.github.io/gamepad/extensions.html
+group: gamepad
```
[the remark this answers is not in the document]
Comment (Elchi3) [reply features/gamepad-extensions.yml:4]: No, this is actually seems to be for the (deprecated) VR spec.

**Comment by Elchi3** [comment]
Thanks for the reviews, everyone. Taking over this work in https://github.com/web-platform-dx/web-features/pull/2434.
